components/Banner/examples/BannerTypicalUseExampleViewController.m

```diff
@@ -160,199 +205,137 @@ + (NSDictionary *)catalogMetadata {
+  [self showSingleLineStyleBanner];
+  self.bannerView.icon = [UIImage imageNamed:@"Email"];
 }
 
 - (void)showMultiLineAlignedButtonStyleBanner {
-  if (self.bannerViewContainer) {
-    [self.bannerViewContainer removeFromSuperview];
+  if (self.bannerView) {
+    [self.bannerView removeFromSuperview];
   }
-  // Set up banner view container
-  UIView *bannerViewContainer = [[UIView alloc] initWithFrame:CGRectZero];
-  [self.view addSubview:bannerViewContainer];
-  bannerViewContainer.backgroundColor = [UIColor whiteColor];
-  self.bannerViewContainer = bannerViewContainer;
 
   MDCBannerView *bannerView = [[MDCBannerView alloc] init];
+  bannerView.preferredContentWidth = exampleBannerContentWidth;
   bannerView.text = exampleLongText;
-  bannerView.image = [UIImage imageNamed:@"Email"];
-  bannerView.numberOfButtons = 2;
+  bannerView.backgroundColor = self.colorScheme.surfaceColor;
+  [self.view addSubview:bannerView];
+  self.bannerView = bannerView;
 
-  UIFont *buttonFont = [MDCTypography body2Font];
-  MDCButton *dismissButton = bannerView.buttons[0];
+  MDCButton *dismissButton = [[MDCButton alloc] initWithFrame:CGRectZero];
   [dismissButton setTitle:@"DISMISS" forState:UIControlStateNormal];
-  [dismissButton setTitleFont:buttonFont forState:UIControlStateNormal];
-  [dismissButton setTitleFont:buttonFont forState:UIControlStateHighlighted];
   [dismissButton setTitleColor:self.colorScheme.primaryColor forState:UIControlStateNormal];
   dismissButton.backgroundColor = self.colorScheme.surfaceColor;
   [dismissButton sizeToFit];
   [dismissButton addTarget:self
                     action:@selector(dismissBanner)
           forControlEvents:UIControlEventTouchUpInside];
-  MDCButton *changeTextButton = bannerView.buttons[1];
-  [changeTextButton setTitle:@"CHANGE TEXT" forState:UIControlStateNormal];
-  [changeTextButton setTitleFont:buttonFont forState:UIControlStateNormal];
-  [changeTextButton setTitleFont:buttonFont forState:UIControlStateHighlighted];
+  MDCButton *changeTextButton = [[MDCButton alloc] initWithFrame:CGRectZero];
+  [changeTextButton setTitle:@"LONG DISMISS" forState:UIControlStateNormal];
   [changeTextButton setTitleColor:self.colorScheme.primaryColor forState:UIControlStateNormal];
   changeTextButton.backgroundColor = self.colorScheme.surfaceColor;
   [changeTextButton sizeToFit];
   [changeTextButton addTarget:self
-                       action:@selector(changeText)
+                       action:@selector(dismissBanner)
              forControlEvents:UIControlEventTouchUpInside];
-
-  self.bannerView = bannerView;
-  [self.bannerViewContainer addSubview:bannerView];
-  CGSize bannerViewSize =
-      [bannerView sizeThatFits:CGSizeMake(exampleBannerContentMultiLineWidth, CGFLOAT_MAX)];
-  bannerView.frame = CGRectMake(0.0f, 0.0f, bannerViewSize.width, bannerViewSize.height);
+  NSMutableArray *buttons = [NSMutableArray arrayWithObjects:dismissButton, changeTextButton, nil];
```

Comment:
Tip: Consider using an array literal rather than nil-terminated variable argument constructors:

```objc
bannerView.buttons = [@[ dismissButton, changeTextButton ] mutableCopy];
```

Reply:
Same as mentioned above if it sounds good.
> The current API of `MDCBannerView` is `NSMutableArray<MDCButton *>`, which led to the code landed here (which I could change to `[@[button] mutableCopy]`. I will make myself note here to use the best/clean way to add buttons based a finalized API after API review if it sounds good to you.